File: NoteApi/src/main/java/lt/viko/eif/mobiliosiosprogrameles/noteapi/model/Note.java
package lt.viko.eif.mobiliosiosprogrameles.noteapi.model;

public class Note {

    private int id;
    private Category category;
    private String title;
    private String text;
    private long date;

    public Note() {}
    public Note(int id, Category category, String title, String text, long date) {
        this.id = id;
        this.title = title;
        this.category = category;
        this.text = text;
        this.date = date;
    }

    public int getId() {
        return id;
    }
    public void setId(int id) {
        this.id = id;
    }

    public Category getCategory() {
        return category;
    }
    public void setCategory(Category category) {
        this.category = category;
    }

    public String getTitle() {
        return title;
    }
    public void setTitle(String title) {
        this.title = title;
    }

    public String getText() {
        return text;
    }
    public void setText(String text) {
        this.text = text;
    }

    public long getDate() {
        return date;
    }
    public void setDate(long date) {
        this.date = date;
    }
}

File: NoteApp/app/src/main/java/lt/viko/eif/mobiliosiosprogrameles/noteapp/domain/model/Note.kt
package lt.viko.eif.mobiliosiosprogrameles.noteapp.domain.model

import lt.viko.eif.mobiliosiosprogrameles.noteapp.data.local.note.NoteEntity
import lt.viko.eif.mobiliosiosprogrameles.noteapp.data.remote.note.NoteDto

data class Note(
    val id: Int? = null,
    var category: Category? = null,
    val title: String,
    val text: String,
    val date: Long,
) {
    fun toNoteEntity(): NoteEntity {
        return NoteEntity(
            noteId = id,
            category = category!!,
            title = title,
            text = text,
            date = date
        )
    }

    fun toNoteDto(): NoteDto {
        return NoteDto(
            id = id,
            category = category!!.toCategoryDto(),
            title = title,
            text = text,
            date = date
        )
    }
}

class InvalidNoteException(message: String) : Exception(message)